frontend/apps/monkvision/js/main.mjs

```diff
@@ -17,11 +17,22 @@ const SELECTED_DATES = "__monkvision_selecteddates", DASHBOARD_TIMER = "__monkvi
 
 const dateAsHTMLDateValue = date => new Date(date.toString().split('GMT')[0]+' UTC').toISOString().split('.')[0];
 
-function timeRangeUpdated(stopRefresh, dates) {
```

Comment:
This function used to receive dates from the caller. Now it does not. How does that impact the use cases?

Reply:
Previously, the function received dates and updated the date picker values. If no dates were provided, the default would be 1 week for the date picker. The new function receives the selected interval value and updates the date picker values when a time interval is selected from the dropdown. It also handles direct changes to the date pickers in the else part, updating the dates accordingly. Overall, the approach to updating the date values has been changed to handle this additional condition.